src/UserApplication/Controllers/ValuesController.cs

```diff
@@ -31,5 +32,73 @@ public async Task<ActionResult<string>> GetValues(string id)
 
             return $"Id value from URL: {id}";
         }
+
+        /// <summary>
+        /// Create new user
+        /// </summary>
+        /// <param name="user"></param>
+        /// <returns></returns>
+        [HttpPost()]
+        public Task<ActionResult<string>> CreateNewUser(UserInfo user)
+        {
+
+            throw new NotImplementedException();
+
+        }
+
+        /// <summary>
+        /// Delete user
+        /// This should only delete user information.
+        /// This shouldn't delete any notes or comments.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpDelete("{id}")]
+        public Task<ActionResult<string>> DeleteUser(string id)
+        {
+
+            throw new NotImplementedException();
+
+        }
+
+        /// <summary>
+        /// Updates a User's username
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpPut("{id}/{username}")]
+        public Task<ActionResult<string>> UpdateUserName(string id)

```

Comment:
These update methods can be merged into a single one. UpdateUser should look a lot like CreateUser except you would pass an id of the user you would like to update.

For example. What if we decide that the user should also have an age which can be changed in the future for whatever reason. Not only would we need to change the model, we would need to write more methods in the controllers to handle that. 

Suggestion: I think changing the signature from `UpdateUser(string id)` to `UpdateUser(string id, UserInfo userInfoUpdate)` would be the way to go. That way, if we add an attribute, it's okay. It's already handled